src/dmd/dinterpret.d

```diff
@@ -77,13 +78,21 @@ public Expression ctfeInterpret(Expression e)
     if (e.type.ty == Terror)
         return new ErrorExp();
 
+    auto g = &ctfeGlobals;
+    ++g.count;
+
     Expression result = interpret(e, null);
 
+    result = copyRegionExp(result);
+
     if (!CTFEExp.isCantExp(result))
         result = scrubReturnValue(e.loc, result);
     if (CTFEExp.isCantExp(result))
         result = new ErrorExp();
 
+    if (--g.count == 0)         // if no more users of g.region
+        g.region.release();     // release all memory in the region
```

Comment:
Why wait till zero? I had imagined that CTFE memory regions could be stacked.

Reply:
The interpreter does nested calls to interpret, as it can call semantic() on code it needs to interpret, which can also interpret. This is more than just stacking.

Reply:
But as the result Expression of a nested call to interpret is memcpy()'d out, isn't it safe to discard any other temporary memory that may have occurred along the way?